### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by Mesi21**
- Added models for user, comment, post, and like.

- Added a method for the user to return the 3 most recent posts of the user.

- Added a method for the post to update the posts counter and one, which returns the 5 most recent comments for a given post.

- Added a method for the comments to update the comments counter for a certain post.

- Added a method for the likes to update the likes counter for a certain post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  has_many :posts
```
Comment (brainconnect93) [inline app/models/user.rb:3]: - [x] Great effort working on your project, according to the ERD diagram, user `has_many likes and comments` too

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  validates :text, presence: true
+  belongs_to :user
+  belongs_to :post
+
+  private
+
+  def update_counter_for_comments
```
Comment (brainconnect93) [inline app/models/comment.rb:9]: - [x] Great effort working on your project, kindly give the `update_counter_for_comment` the after_save method just the way you gave the likes and posts model

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @Mesi21, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 
- Descriptive Pull Request title and summary. ✔️  
- Blog App - processing data in models :heavy_check_mark: 
- Good use of Git flow ✔️ 

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Mesi21 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the suggested changes from the previous reviewer. 👍 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**